### PR TITLE
feat: make file download dialog component independent from parent

### LINKED ISSUE #1003: Cannot cancel GCode preview loading when using "Automatically load file on print start"
### Fluidd Version

1.22.2

### Browser

Chrome

### Device

Desktop PC, Mobile device (Phone, Tablet or similar)

### Operating System

Windows, Android

### What happened

I am using GCode preview and I have enabled "Automatically load file on print start".
Sometimes when I'm checking from my phone using mobile data, I want to cancel that preview download, because my data plan has limited speed and I only want to quickly check the webcam. However it's currently impossible to cancel it, the dialog just reappears and the download continues (with NaN as remaining download size).

https://user-images.githubusercontent.com/13891066/210394523-045dd0b8-5121-4aa7-a83d-59456dbe36c8.mp4

### What did you expect to happen

Downloading should stop

### How to reproduce

- Enable "Automatically load file on print start"
- Open Fluidd
- Try to cancel the preview download (you can use Chrome network throttling for better reproduction)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by kieraneglin**
@pedrolamas informed me of an issue where the file download dialog wouldn't appear if the `Jobs` (or, really, the `FileSystem`) component wasn't actively rendered on-page.

This PR refactors the way the file download dialog is interacted with by making it independent of any specific parent component. To accomplish this, I also had to refactor the way Axios `cancelTokenSource`'s were created/referenced by adding them into Vuex.

Fixes #1003

Signed-off-by: Kieran Eglin <kieran.eglin@gmail.com>

### CODE REVIEW

File: src/store/files/types.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import { KlipperFileMeta, Thumbnail } from './types.metadata'
 import { HistoryItem } from '@/store/history/types'
+import { CancelTokenSource } from 'axios'
 
 export type { KlipperFileMeta, Thumbnail }
 
 export interface FilesState {
   uploads: FilesUpload[];
   download: FileDownload | null;
+  fileTransferCancelTokenSource: CancelTokenSource | null;
```
Comment (kieraneglin) [inline src/store/files/types.ts:10]: This is a mouthful so I'm more than open to naming suggestions!

File: src/views/GcodePreview.vue [hunk 2]
```diff
@@ -1,14 +1,22 @@
 <template>
-  <GcodePreviewCard full-screen />
+  <v-row :dense="$vuetify.breakpoint.smAndDown">
+    <v-col cols="12">
+      <GcodePreviewCard full-screen />
+    </v-col>
+
+    <file-system-download-dialog />
```
[the remark this answers is not in the document]
Comment (kieraneglin) [reply src/views/GcodePreview.vue]: Good callout! I've made that change.

My initial reason to put this here was to get the download dialog on the standalone gcode page when you click "Load current file". In truth, I didn't test to see if including this component on the dashboard also made it work here

File: src/store/files/types.ts (hunk 1 above)
Comment (pedrolamas) [reply src/store/files/types.ts:10]: Naming is meh... but honestly, it describes what it is perfectly and that is not visible to the user, so I don't really care! 😁

**Review by matmen — APPROVED** [review]
LGTM, thanks!

**Pull request merged**